frontend/src/features/public-form/components/PublicSwitchEnvMessage.tsx

```diff
@@ -16,9 +16,19 @@ export const PublicSwitchEnvMessage = (): JSX.Element => {
           mb="1.5rem"
           mt={{ base: '2rem', md: '0' }}
         >
+          {/* Hide the text and replicate it in the aria-label for the button instead,
+           because without it, the sentence will be read as two separate parts which
+          is confusing for those using screen readers. */}
           <Text>
-            You’re using the new FormSG design. If you have trouble submitting,
-            <Button variant="link" onClick={onOpen}>
+            <span aria-hidden>
+              You’re using the new FormSG design. If you have trouble
+              submitting,
+            </span>
+            <Button
+              variant="link"
+              onClick={onOpen}
+              aria-label="You're using the new FormSG design. If you have trouble submitting, switch to the original one here."
+            >
```

Comment:
```suggestion
            <HStack spacing="0px">
            <Text aria-hidden>
              You’re using the new FormSG design. If you have trouble
              submitting,
            </Text>
            <Button
              variant="link"
              onClick={onOpen}
              aria-label="You're using the new FormSG design. If you have trouble submitting, switch to the original one here."
            >
              <Text as="u">switch to the original one here.</Text>
            </Button>
          </HStack>
```

To keep it standardised should we stick to Chakra instead of `span`? Don't think we have `span` anywhere else in the react app